### ISSUE TITLE
Fix S2077 FN track concatenated values

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [SonarSource/sonar-dotnet#3292]
With https://github.com/SonarSource/sonar-dotnet/pull/3106 variables with string interpolation values "are tracked" :metal:

```
string sensitiveQuery = string.Format("INSERT INTO Users (name) VALUES (\"{0}\")", param); //  SQL query with string interpolation
string oString3 = sensitiveQuery; // assignment
SqlCommand command = new SqlCommand(oString3); // Noncompliant
command.CommandText = oString3; // Noncompliant
```

However variables with string concatenation values and used in sql queries do not trigger the rule: 
```
string oString2 = "SELECT * FROM Table1 WHERE col1 = '" + param + "'"; //  SQL query with string concatenation
SqlCommand oCmd2 = new SqlCommand(oString2, myConnection); // FN
oCmd2.CommandText = oString2; // FN
```

Related community thread:
https://community.sonarsource.com/t/owasp-rules-not-applied-on-c-and-vb-net-code/22919

Eric
